### ISSUE TITLE
Autocomplete suggestions don't work with Track Changes

### ISSUE THREAD

**Issue by Mgsy** [WebSpellChecker/wproofreader-ckeditor5#62]
## Steps to reproduce

1.  Open the editor with Track Changes and WProofreader.
2.  Enable _Autocomplete suggestions_ in WProofreader options.
3.  Enable Track Changes.
4.  Start typing.

## Expected result

The text suggestion appears.

## Current result

The text suggestion doesn't appear.

## Notes

It works fine in the following scenario:

1.  Open the editor with Track Changes and WProofreader.
2.  Enable _Autocomplete suggestions_ in WProofreader options.
3.  Enable Track Changes.
4.  Put the caret at the end of a paragraph.
5.  Press enter to create a new paragraph.
6.  Move back to the end of the paragraph from step 4.
7.  Start typing.

**Comment by okulykova** [comment]
@Mgsy Thank you for providing the detailed steps to reproduce the issue.

Could you please specify which CKEditor 5 version you are referring to? Also, let us know if you faced such an issue in any of the previous CKEditor 5 versions.

After you provide additional information, our technical team will be able to review it in more detail.

Thanks.

**Comment by Mgsy** [comment]
Hi @okulykova! This issue could be reproduced in the latest version (v36.0.1), but I was able to reproduce it also in older version, e.g. v35.2.1.

**Comment by okulykova** [comment]
@Mgsy Thank you for sharing additional information. Our technical specialists reviewed it.

The mentioned behavior happens because when the Track Changes plugin is used, seven hidden DOM elements **&NoBreak;** are added to the text. It affects WebSpellChecker internal mechanisms. Due to the number of hidden symbols which the server can't interpret, it doesn't generate suggestions and returns an empty response.

We've added this issue to our backlog for further investigation.

**Comment by Forium777** [comment]
Hello @okulykova ! I hope you're doing well. I've a question, why the fuck are you using right arrow to accept suggestion ? Don't you see it is far away from the part of the keyboard we're mainly using? Don't you think there is a lot of keyboard without arrows/numpad? Don't you see Gmail is using TAB key ??

Thanks.

**Comment by okulykova** [comment]
Hello @Forium777. Thank you for your feedback. We are aware that the right arrow is placed  aside from the main keyboard. We are also aware that there is a limited number of keyboards that don't have any arrows at all. As usual, these are gaming keyboards and the work in Excel, for example, may also be complicated for this reason.

There are multiple approaches for autocomplete mechanism implementation and, apart from the right arrow, the Tab key is also used in some cases. These are cases when there is a native autocomplete mechanism which is under the full control of its developers. However, in our case, WebSpellChecker products are integrated into different environments (e.g. WYSIWYG editors like CKEditor) that are not under our control. The Tab key usually plays a specific role there and we can't just cancel or override its already existing action.

Anyway, as a part of integration with CKEditor 5 specifically, we are considering the possibility of using other keys and already created the task for it.

**Could you please specify where you are using the WebSpellChecker product (HTML editable elements, the specific WYSIWYG editor like CKEditor, Froala Editor, ...) or if you are using the browser extension?**

**Comment by Forium777** [comment]
No worries, I'm using it under CKEditor but it does not matter actually. You could just provide an option [here](https://webspellchecker.com/docs/api/wscbundle/Options.html) to choose for a key (some people could except SPACE to autocomplete). CKEditor is not using TAB either way, even if it'd you'd handling key conflict would not be that difficult.

When can we except that?

**Comment by okulykova** [comment]
@Forium777, it is already in our plans to implement the apply key/button change via the mentioned options as you mentioned. However, there is no ETA yet. As soon as we implement such an option, we will let you know.

As for now, you can disable the autocomplete suggestions feature through UI by turning off the "Autocomplete suggestions" option in the settings menu. Also, you can disable it in the WProofreading config by adding the following.

**autocomplete: false,**

[Here](https://docs.webspellchecker.com/display/WebSpellCheckerCloud/Autocomplete+suggestions) you can find more details.

**Comment by okulykova** [comment]
@Forium777, we are glad to let you know that we've completed the task to implement the possibility to accept the autocompete suggestions by using the "Tab" key on the keyboard. Now both options, the "Tab"key and/or "Right arrow", are applicable to accept autocomplete suggestions. 

This option is available in our latest version, which has already been released. Cloud clients get the upgrade automatically (no additional actions are needed).
